apps/src/StudioApp.js

```diff
@@ -1524,15 +1524,21 @@ StudioApp.prototype.resizeVisualization = function(width) {
  */
 StudioApp.prototype.resizeToolboxHeader = function() {
   var toolboxWidth = 0;
-  if (
-    this.editCode &&
-    this.editor &&
-    this.editor.session &&
-    this.editor.session.paletteEnabled
-  ) {
-    // If in the droplet editor, set toolboxWidth based on the block palette width:
-    var categories = document.querySelector('.droplet-palette-wrapper');
-    toolboxWidth = categories.getBoundingClientRect().width;
+  if (this.editCode && this.editor && this.editor.session) {
+    const isRtl = getStore().getState().isRtl;
+    const categories = document.querySelector('.droplet-palette-wrapper');
+
+    if (this.editor.session.paletteEnabled) {
+      // If in the droplet editor, set toolboxWidth based on the block palette width:
+      toolboxWidth = categories.getBoundingClientRect().width;
+    }
+
+    if (isRtl) {
+      // If Rtl - handle show/hide toolbox functionality
+      categories.style.zIndex = this.editor.session.paletteEnabled
+        ? 'inherit'
+        : '0';
+    }
```

Comment:
nit: variables should be initialized when are going to be closed to be used
```
    if (this.editor.session.paletteEnabled) {
      // If in the droplet editor, set toolboxWidth based on the block palette width:
      toolboxWidth = categories.getBoundingClientRect().width;
    }
   const isRtl = getStore().getState().isRtl;
    const categories = document.querySelector('.droplet-palette-wrapper');

    if (isRtl) {
      // If Rtl - handle show/hide toolbox functionality
      categories.style.zIndex = this.editor.session.paletteEnabled
        ? 'inherit'
        : '0';
    } 
```

Reply:
1.  const categories - We need it in the first If statement, so they can't be declared later
2. const rtl - is more of a holy war object I guess. I prefer to keep all the variables declarations in the beginning of the function/block scope so that code stays easy to read and understand